File: server/guess/src/main/java/com/alloymobile/guess/persistence/dbo/Word.java
package com.alloymobile.guess.persistence.dbo;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Data
@ToString(exclude = {"categoryWord"})
public class Word implements IGuessDBO {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @ManyToOne(fetch = FetchType.LAZY, cascade = {CascadeType.PERSIST,CascadeType.MERGE})
    @JoinColumn(name="CATEGORY_ID")
    private Category categoryWord;
}

File: client/guess/src/app/model/word.ts
import { Category } from './category';

export class Word {
    id: number;
    name: string;
    category: Category;
    constructor(word?: Word){
        if (word) {
            this.id = word.id;
            this.name = word.name;
            this.category = new Category(word.category);
        } else {
            this.id = 0;
            this.name = '';
            this.category = new Category();
        }
    }
}
